src/protocol/sort/reshare.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{
+    error::BoxError,
+    field::Field,
+    helpers::{
+        mesh::{Gateway, Mesh},
+        prss::SpaceIndex,
+        Direction, Identity,
+    },
+    protocol::{context::ProtocolContext, RecordId, Step},
+    secret_sharing::Replicated,
+};
+use embed_doc_image::embed_doc_image;
+use serde::{Deserialize, Serialize};
+
+/// A message sent by each helper when they've reshared their own shares
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct PartValue<F> {
+    part: F,
+}
+/// Reshare(i, [x])
+// This implements reshare algorithm of "Efficient Secure Three-Party Sorting Protocol with an Honest Majority" at communication cost of 2R.
+// Input: Pi-1 and Pi+1 knows their secret shares
+// Output: At the end of the protocol, [all 3 helpers receive renewed shares of x with randomness added
+#[derive(Debug)]
+pub struct Reshare<F> {
+    input: Replicated<F>,
+}
+
+impl<F: Field> Reshare<F> {
+    #[allow(dead_code)]
+    pub fn new(input: Replicated<F>) -> Self {
+        Self { input }
+    }
+
+    /// Steps
+    #[embed_doc_image("reshare", "images/sort/reshare.png")]
+    /// 1. While calculating for helper, we call PRSS to get randoms for helper who needs to receive reshares (say `rand_left`, `rand_right`)
+    ///    `to_helper.left` knows `rand_left` (named r1) and `to_helper.right` knows `rand_right` (named r0)
+    /// 2. `to_helper.left` calculates part1 = (input.0 + input.1) - r1
+    ///    `to_helper.right` calculates part2 = (input.0 - r0)
```

Comment:
based on the image, I assume you mean input.2 and not input.0

Reply:
Thats right but from to_helper.right POV, his share corresponds to input.0. This comment is to ensure it matches with what is coded below